tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -1481,11 +1478,11 @@ Status StorageManagerCanonical::group_create(const std::string& group_uri) {
 }
 
 ThreadPool* StorageManagerCanonical::compute_tp() const {
```

Comment:
These accessor functions should have been declared `inline` in the header a long time ago. This is a good opportunity to move them.

While we want to have `ContextResources` return references from its accessors, we should leave the ones here alone and have them continue to return pointers. It's too much text to change, and for many of the changes we want to convert to using `ContextResources`, which is the right time to adjust pointers to references.

Reply:
Moved all of the related accessors to be `[[nodiscard]] inline` in their respective headers.